presto-main/src/main/java/io/prestosql/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1058,10 +1144,28 @@ protected Scope visitTable(Table table, Optional<Scope> scope)
             QualifiedObjectName name = createQualifiedObjectName(session, table, table.getName());
             analysis.addEmptyColumnReferencesForTable(accessControl, session.getIdentity(), name);
 
-            // is this a reference to a view?
-            Optional<ConnectorViewDefinition> optionalView = metadata.getView(session, name);
-            if (optionalView.isPresent()) {
-                return createScopeForView(table, name, scope, optionalView.get());
+            // If this is a materialized view, get the name of the storage table
+            Optional<QualifiedName> storageName = getMaterializedViewStorageTableName(table, name);
+            Optional<TableHandle> storageHandle = Optional.empty();
+            if (storageName.isPresent()) {
+                storageHandle = getMaterializedViewStorageTableHandle(table, storageName.get());
+            }
+
+            // If materialized view is current, answer the query using the storage table
+            Identifier catalogName = new Identifier(name.getCatalogName());
+            Identifier schemaName = new Identifier(name.getSchemaName());
+            Identifier tableName = new Identifier(name.getObjectName());
+            QualifiedName mvName = QualifiedName.of(ImmutableList.of(catalogName, schemaName, tableName));
+            Optional<TableHandle> mvHandle = metadata.getTableHandle(session, createQualifiedObjectName(session, table, mvName));
+            if (storageHandle.isPresent() && metadata.isMaterializedViewCurrent(session, mvHandle.get()).getKey() == true) {
+                name = createQualifiedObjectName(session, table, storageName.get());
```

Comment:
We already have the storage table handle at this point. No need to roundtrip again via the table name.

Reply:
I moved the evaluation of 'tableHandle' up and simply assigned 'storageHandle' to 'tableHandle' at this point.